### PR TITLE
Improved TFN and MACE implementation

### PR DESCRIPTION
**Pull request by chaitjo**
This PR implements improvements to the TFN model in order to benchmark it.
- DiffDock style concatenation of src and dst scalar type features into edge scalar features (which are used to compute the weights of the tensor product in each TFN layer). This would allow the model to learn weights customised to each src-dst type as opposed to one generic set of weights which were previously only dependent on the radial filter.
- Revert to gaussian RBF function, as the Bessel function comes from molecular dynamics models like MACE/DimeNet, but it may not be suitable for general purpose tasks. The Bessel function penalises far away neighbours' information very heavily.

This PR also makes improvements to the EGNN model for stable training, including:
- Changing to mean global pooling, as well as batch norm.
- Adding dropouts.
- Adding displacement scaling by distances to normalise the positional update.

Please do not merge this PR yet. This is still a work in progress and some unresolved aspects remain:
- Whether to use an O(3) equivariant model or an SO(3) equivariant one? (This can be customised fully by the user now, using the `hidden_irreps` parameter when initialising the model)
    - **Decision:** we will go with SO(3) and higher order tensors, as it is interesting to investigate their impact and is the correct inductive bias for proteins.
- Implement the same updates to the MACE model, too.
- ~~Whether we should move to a new approach for creating edges, where we explicitly call `to_undirected` after creating the `edge_index` with knn or eps methods.~~

### CODE REVIEW

File: proteinworkshop/models/graph_encoders/tfn.py [hunk 1]
```diff
@@ -169,9 +151,7 @@ def required_batch_attributes(self) -> Set[str]:
     @jaxtyped
     @beartype
     def forward(self, batch: Union[Batch, ProteinBatch]) -> EncoderOutput:
-        """Implements the forward pass of the TFN encoder.
-
-        Returns the node embedding and graph embedding in a dictionary.
+        """Returns the node embedding and graph embedding in a dictionary.
```
Comment (amorehead) [inline proteinworkshop/models/graph_encoders/tfn.py:154]: I think we are now using standardized language for each method's forward function docstring. If you would be able to update this docstring to match the others, that would be great.

**Comment by amorehead** [comment]
Regarding the default option for equivariance, I lean more towards defaulting to SO(3) equivariance, since a lot of protein-centric work tends to favor this kind of inductive bias regarding 3D reflections (at least when it comes to chiral biomolecules).

File: proteinworkshop/config/encoder/egnn.yaml [hunk 2]
```diff
@@ -2,7 +2,8 @@ _target_: "proteinworkshop.models.graph_encoders.egnn.EGNNModel"
 num_layers: 6 # Number of message passing layers
 emb_dim: 512 # Dimension of the node embeddings
 activation: relu # Activation function to use
-norm: layer # Normalisation layer to use
+norm: batch # Normalisation layer to use
 aggr: "sum" # Aggregation function to use
-pool: "sum" # Pooling operation to use
+pool: "mean" # Pooling operation to use
 residual: True # Whether to use residual connections
+drop_rate: 0.1 # Dropout rate
```
Comment (a-r-j) [inline proteinworkshop/config/encoder/egnn.yaml:9]: Can we rename this to `dropout` for consistency with other models?

File: proteinworkshop/config/encoder/tfn.yaml [hunk 3]
```diff
@@ -1,13 +1,13 @@
...
-emb_dim: 64 # Number of hidden channels/embedding dimension for each node feature tensor order
+hidden_irreps: "32x0e + 32x0o + 8x1e + 8x1o + 4x2e + 4x2o" # Irreps string for intermediate layer node feature tensors
 mlp_dim: 256 # Dimension of MLP for computing tensor product weights
 aggr: "sum" # Aggregation function to use
-pool: "sum" # Pooling operation to use
+pool: "mean" # Pooling operation to use
 residual: True # Whether to use residual connections
 batch_norm: True # Whether to use e3nn batch normalization
 gate: False # Whether to use gated non-linearity
+drop_rate: 0.1 # Dropout rate
```
Comment (a-r-j) [inline proteinworkshop/config/encoder/tfn.yaml:13]: as above

File: proteinworkshop/models/graph_encoders/tfn.py (hunk 1 above)
Comment (chaitjo) [reply proteinworkshop/models/graph_encoders/tfn.py]: Done

File: proteinworkshop/config/encoder/egnn.yaml (hunk 2 above)
Comment (chaitjo) [reply proteinworkshop/config/encoder/egnn.yaml]: Fixed
